File: ts/medium/simpleQueue.ts
function main() {
  class Queue {
    private _queue: string[] = [];

    private _isInQueue(item: string): boolean {
      for (const elem of this._queue) {
        if (item.toLowerCase() === elem.toLowerCase()) {
          return true;
        }
      }
      return false;
    }

    get queue(): string[] {
      return this._queue;
    }

    enqueue(item: string): void {
      if (!this._isInQueue(item)) {
        this._queue.push(item);
      }
    }
    
    dequeue(): void {
      if (this._queue.length) {
        this._queue.shift();
      }
    }
  }

  const q = new Queue();
  console.log(q.queue);
  q.enqueue("Jack");
  console.log(q.queue);
  q.enqueue("Mag");
  console.log(q.queue);
  q.dequeue();
  console.log(q.queue);
}
main();

File: python/medium/simple_queue.py
class Queue:
    def __init__(self):
        self.queue = list()

    def enqueue(self, item):
        if item not in self.queue:
            self.queue.append(item)

    def dequeue(self):
        if len(self.queue):
            self.queue = self.queue[1:]


    def get_queue(self):
        return self.queue

def main():
    q = Queue()
    q.enqueue("Jack")
    q.enqueue("Mag")
    print(q.get_queue())
    q.dequeue()
    print(q.get_queue())

if __name__ == "__main__":
    main()